File: students/mark_pereira/MTA.rb
require 'pry'

def menu
subway_lines = { 
	"line_N" => ["Time Square", "34th", "28th", "23rd", "Union Square", "8th"],
	"line_L" => ["8th", "6th", "Union Square", "3rd", "1st"],
	"line_6" => ["Grand Central", "33rd", "28th", "23rd", "Union Square", "Astor Place"]
	}

puts "Which line are you on?"
puts "#{subway_lines.keys}"

choice_line = gets.chomp

loop do
	if subway_lines.keys.include?choice_line 
		break
	else
		puts " This is not a valid train line. Please re-enter your choice"
		choice_line = gets.chomp
	end
end
		

puts "Please select the station you want to get on at:"
puts "#{subway_lines[choice_line]}"

choice_station = gets.chomp

loop do
	if subway_lines[choice_line].include?choice_station
		break
	else
		puts "This is not a valid station. Please re-enter your choice"
		choice_station = gets.chomp
	end
end

puts "Please select the final destination:"
puts "#{subway_lines.values}"

choice_destination = gets.chomp

# loop do
# 	if subway_lines.values.split.include?choice_destination
# 		break
# 	else
# 		puts "This is not a valid station. Please re-enter your choice"
# 		choice_destination = gets.chomp
# 	end
# end
# binding.pry

puts "Which line is your final destination on?"
puts "#{subway_lines.keys}"

choice_final_line = gets.chomp

loop do
	if subway_lines.keys.include?choice_final_line
		break
	else
		puts "This is not a valid destination. Please re-enter your choice"
		choice_final_line = gets.chomp
	end
end


if choice_line == choice_final_line
	
	stops = subway_lines[choice_final_line].index(choice_destination) - subway_lines[choice_line].index(choice_station) 
	puts "#{stops.abs}"
else
	stop1 = subway_lines[choice_line].index("Union Square")-subway_lines[choice_line].index(choice_station)
	
	stop2 = subway_lines[choice_final_line].index("Union Square") - subway_lines[choice_final_line].index(choice_destination)
	
	puts "You have #{stop1.abs + stop2.abs} stops from your start point: #{choice_station} Station (#{choice_line}) to your destination point: #{choice_destination} Station (#{choice_final_line})."
 end
end

menu()

File: students/charlie_gerard/javascript_homework/MTA/js/mta.js
var trip = function(){
	var lines = {
		lineN: ["Time Square", "34th", "28th_lineN", "23rd_lineN", "Union Square", "8th_lineN"],
		lineL: ["8th_lineL", "6th", "Union Square", "3rd", "1st"],
		line6: ["Grand Central", "33rd", "28th_line6", "23rd_line6", "Union Square", "Astor Place"]
	};

	var choice = [
		start_line = Object.keys(lines)[0],
		start_station = lines.lineN[0],
		end_line = Object.keys(lines)[2],
		end_station = lines.line6[0]
	];

	return {
		numStops: function(){
			if(start_line === end_line){
				//Finds the index of the starting and ending station on the line.
				return Math.abs(lines.start_line.indexOf(start_station)) - Math.abs(lines.end_line.indexOf(end_station));
			} else {
				//Calculate number of stops on starting line
				first_line = Math.abs(lines[start_line].indexOf("Union Square")) - Math.abs(lines[start_line].indexOf(start_station));
				//Calculate number of stops on finish line
				second_line = Math.abs(lines[end_line].indexOf("Union Square")) - Math.abs(lines[end_line].indexOf(end_station));
				stops = first_line + second_line;
				return stops;	
			}
		},

		listStops: function(){
		 	for(i = 0; i <= lines[start_line].indexOf("Union Square"); i++){
		 		stops_first_line = (lines[start_line][i]);
		 		console.log(stops_first_line);
		 	}
		 	//I don't know how to reverse the order of the stations if the users go the other way on the second line
		 	for(i = 0; i < lines[end_line].indexOf("Union Square");i++){
		 		stops_end_line = (lines[end_line][i]);
		 		console.log(stops_end_line);
		 	}
		}
	};
};

var firstTrip = trip();

firstTrip.numStops();

console.log("You are going from " + start_station + " on " + start_line + " to " + end_station + " on " + end_line + ".");
console.log("You have " + firstTrip.numStops() + " stops between " + start_station + " and " + end_station + ".");
console.log("Please find below a list of the stations for your trip: ");
firstTrip.listStops();
